src/managers/ApplicationCommandManager.js

```diff
@@ -0,0 +1,307 @@
+   * @example
+   * // Fetch all commands
+   * guild.commands.fetch()
+   *   .then(commands => console.log(`Fetched ${commands.size} commands`))
+   *   .catch(console.error);
+   */
+  async fetch(id, cache = true, force = false) {
+    if (id) {
+      if (!force) {
+        const existing = this.cache.get(id);
+        if (existing) return existing;
+      }
+      const command = await this.commandPath(id).get();
+      return this.add(command, cache);
+    }
+
+    const data = await this.commandPath.get();
+    return data.reduce((coll, command) => coll.set(command.id, this.add(command, cache)), new Collection());
+  }
+
+  /**
+   * Creates an application command.
+   * @param {ApplicationCommandData} command The command
+   * @returns {Promise<ApplicationCommand>}
+   * @example
+   * // Create a new command
+   * client.application.commands.create({
+   *   name: 'test',
+   *   description: 'A test command',
+   * })
+   *   .then(console.log)
+   *   .catch(console.error);
+   */
+  async create(command) {
+    const data = await this.commandPath.post({
+      data: this.constructor.transformCommand(command),
+    });
+    return this.add(data);
+  }
+
+  /**
+   * Sets all the commands for this application or guild.
+   * @param {ApplicationCommandData[]} commands The commands
+   * @returns {Promise<Collection<Snowflake, ApplicationCommand>>}
+   * @example
+   * // Set all commands to just this one
+   * client.application.commands.set([
+   *   {
+   *     name: 'test',
+   *     description: 'A test command',
+   *   },
+   * ])
+   *   .then(console.log)
+   *   .catch(console.error);
+   * @example
+   * // Remove all commands
+   * guild.commands.set([])
+   *   .then(console.log)
+   *   .catch(console.error);
+   */
+  async set(commands) {
+    const data = await this.commandPath.put({
+      data: commands.map(c => this.constructor.transformCommand(c)),
+    });
+    return data.reduce((coll, command) => coll.set(command.id, this.add(command)), new Collection());
+  }
+
+  /**
+   * Edits an application command.
+   * @param {ApplicationCommandResolvable} command The command to edit
+   * @param {ApplicationCommandData} data The data to update the command with
+   * @returns {Promise<ApplicationCommand>}
+   * @example
+   * // Edit an existing command
+   * client.application.commands.edit('123456789012345678', {
+   *   description: 'New description',
+   * })
+   *   .then(console.log)
+   *   .catch(console.error);
+   */
+  async edit(command, data) {
+    const id = this.resolveID(command);
+    if (!id) throw new TypeError('INVALID_TYPE', 'command', 'ApplicationCommandResolvable');
+
+    const raw = {};
+    if (data.name) raw.name = data.name;
+    if (data.description) raw.description = data.description;
+    if (data.options) raw.options = data.options.map(o => ApplicationCommand.transformOption(o));
+    if (data.defaultPermission) raw.default_permission = data.defaultPermission;
```

Comment:
This will fail if you pass defaultPermission as `false`

Reply:
Wouldn't it be better to drop the `if`'s? There are cases in which we might want to pass undefined/null, and undefined properties don't get serialized in JSON either way.

Reply:
> Wouldn't it be better to drop the if's? There are cases in which we might want to pass undefined/null, and undefined properties don't get serialized in JSON either way.

Sounds good to me, however I'll keep creating a new object from the provided data to filter out invalid properties